indexing-hadoop/src/main/java/io/druid/indexer/InputRowSerde.java

```diff
@@ -63,9 +230,25 @@
       WritableUtils.writeVInt(out, dimList.size());
       if (dimList != null) {
         for (String dim : dimList) {
-          List<String> dimValues = row.getDimension(dim);
-          writeString(dim, out);
-          writeStringArray(dimValues, out);
+          IndexSerdeTypeHelper typeHelper = typeHelperMap.get(dim);
+          if (typeHelper == null) {
+            typeHelper = STRING_HELPER;
+          }
+          try {
+            byte[] value = typeHelper.serialize(row.getRaw(dim));
+
+            writeString(dim, out);
+            WritableUtils.writeVInt(out, typeHelper.getType().ordinal());
```

Comment:
Writing the type ordinal isn't necessary since the reader should know the type from the name of the dim.

Reply:
dropped the type ordinal